src/main/java/com/mathworks/ci/RunMatlabTestsBuilder.java

```diff
@@ -316,19 +274,144 @@ public String constructCommandForTest(String inputArguments) {
 
     // Concatenate the input arguments
     private String getInputArguments() {
-        final String pdfReport = MatlabBuilderConstants.PDF_REPORT + "," + this.getPdfReportChkBx();
-        final String tapResults = MatlabBuilderConstants.TAP_RESULTS + "," + this.getTapChkBx();
-        final String junitResults =
-                MatlabBuilderConstants.JUNIT_RESULTS + "," + this.getJunitChkBx();
-        final String stmResults =
-                MatlabBuilderConstants.STM_RESULTS + "," + this.getStmResultsChkBx();
-        final String coberturaCodeCoverage =
-                MatlabBuilderConstants.COBERTURA_CODE_COVERAGE + "," + this.getCoberturaChkBx();
-        final String coberturaModelCoverage = MatlabBuilderConstants.COBERTURA_MODEL_COVERAGE + ","
-                + this.getModelCoverageChkBx();
-        final String inputArgsToMatlabFcn = pdfReport + "," + tapResults + "," + junitResults + ","
-                + stmResults + "," + coberturaCodeCoverage + "," + coberturaModelCoverage;
-
-        return inputArgsToMatlabFcn;
+        addInputArgs(MatlabBuilderConstants.PDF_REPORT_PATH, getPdfReportFilePath());
+        addInputArgs(MatlabBuilderConstants.TAP_RESULTS_PATH, getTapReportFilePath());
+        addInputArgs(MatlabBuilderConstants.JUNIT_RESULTS_PATH, getJunitReportFilePath());
+        addInputArgs(MatlabBuilderConstants.STM_RESULTS_PATH, getStmResultsFilePath());
+        addInputArgs(MatlabBuilderConstants.COBERTURA_CODE_COVERAGE_PATH,
+                getCoberturaReportFilePath());
+        addInputArgs(MatlabBuilderConstants.COBERTURA_MODEL_COVERAGE_PATH,
+                getModelCoverageFilePath());

```

Comment:
To me this is pointing to some logic that needs to be contained in other classes (not the builder). For example, if the various artifact classes that you link to the checkbox fields had an `appendInputArguments` method, or a `contributeTo(ScratchInputArguments args)` then each of the artifact classes could actually do this work, and the bud,er wouldn't need to. Also, this would allow moving all these constants to the actual artifact classes you have created, which seem like the right location for that knowledge anyway.

Finally, this would allow you to leverage the NullObject pattern so that you didn't need any null checks. Instead, the default value for the artifacts would be a NullArtifact which would have a contributeTo (or whatever its called) method that simply does nothing.

Reply:
Added addFilePathTo() method for each artifact class and moved all the constants to the specific artifact classes. However i am facing few issues to leverage the same for NullObject pattern. the major problem is the checkboxes are cheked on UI when its not NULL and in the null patteern if nothing is set it still NullArtifact by default hence on UI it appears all the values as checked I might need to discuss with you or team to see how i could handle this using NullObject pattern . meanwhile i hav eupdated the code for isolating the  addFilePathTo() method to each artifact class however here i again will hav eto check for the NULL object anyways.